src/makeInfinityAdder.js

```diff
@@ -6,7 +6,26 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let summary = 0;
+
+  function adder(x) {
+    if (x !== undefined) {
+      return (y) => {
+        if (y !== undefined) {
+          summary += x + y;
+
+          return adder(x + y);
+        }
+        summary = 0;
+
+        return x;
+      };
+    }
+
+    return summary;
```

Comment:
And if `x === undefined`, you should return the final sum and reset summary variable